payjoin/src/core/uri/url_ext.rs

```diff
@@ -125,21 +143,32 @@ where
 
 fn set_param(url: &mut Url, prefix: &str, param: &str) {
     let fragment = url.fragment().unwrap_or("");
-    let mut fragment = fragment.to_string();
-    if let Some(start) = fragment.find(prefix) {
-        let end = fragment[start..].find('+').map_or(fragment.len(), |i| start + i);
-        fragment.replace_range(start..end, "");
-        if fragment.ends_with('+') {
-            fragment.pop();
-        }
-    }
 
-    if !fragment.is_empty() {
-        fragment.push('+');
+    // See above for `-` vs `+` backwards compatibility
+    let mut delim = '-';
+    if !fragment.contains(delim) {
+        delim = '+';
     }
```

Comment:
although very unlikely , if the url contains both "+" and "-" as delimeter , the behaviour becomes confusing , will an else block handle such scenario ?

Reply:
Thanks for the clarification